test/test.c

```diff
@@ -391,7 +391,7 @@ static void test_natsBuffer(void)
 
     test("Changing backend is reflected in buffer: ");
     testCond((s == NATS_OK)
-             && (backend[1] = 'x')
+             && (backend[1] == 'x')
```

Comment:
No, the intent was actually, as the text in `test()` above suggests, to change the backend content and verify that it is reflected in the natsBuffer object.
But I agree that some compilers may flag this as suspicious, so you could move setting `backend[1] = 'x';` above the `testCond()` to make that clearer.

PS: technically the `(s == NATS_OK)` is not required here. This is the remaining of original test code that would not exit the function on a failed `testCond()`. Since I do now, there are cases where checking for `s == NATS_OK` is not required, but it does not hurt, so I would not go over all test.c to remove the ones that are not needed :-).